lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -1494,6 +1494,22 @@ GraphOperationInfo::decodeAttributeName(Identifier name) {
   return { nameStr.substr(0, dollarLoc), opClass };
 }
 
+int GraphOperationInfo::getIntAttr(unsigned attrIdx, StringRef attrName) const {
```

Comment:
Why specify the name and the index, instead of just the name, and using that to find the right index?  GraphOpInst has a "getAttributeNamed" method, maybe these can be hoisted up to it, so you could just use something like:

auto attr = inst->getAttributeNamed("foo", SymbolicValue::Integer);
if (attr) { ... }

and we wouldn't need these helpers?

Reply:
also, if this method does stay, please keep it returning int64_t.

Reply:
Changed to int64_t.

A few reasons for having this API:
1. GraphOperationInst::getAttribute(name) (or the renamed version) uses a loop to find such an attr name -- it could be changed to a hash table in the future though.
2. For synthesized attrs (like tensor ids from sends / recvs and device partitioning), the code would be more robust (but also more rigid admittedly), if the caller gets to assert that no only an attr is present, but it must be in a certain position / idx.